src/charm.py

```diff
@@ -131,20 +134,25 @@ def _reconcile(self) -> None:
         config = CharmConfig.from_charm(self, client)
 
         gateway_resource_definition = GatewayResourceDefinition.from_charm(self)
-        # This line is currently here to validate the existence of the certificates relation.
-        # This charm state component will be used by the upcoming SecretResourceManager.
-        TLSInformation.from_charm(self)
+        tls_information = TLSInformation.from_charm(self)
+        secret_resource_definition = SecretResourceDefinition.from_charm(self)
```

Comment:
Is it still a secret resource definition ? Since it doesn't contain anymore the data allowed to fully create a Secret Resource

I'm wondering if we shouldn't rename the `State` as `ResourceDefinition` and that one differently

Reply:
Yep, I renamed this to `SecretResourceInformation` and the class used in resource_manager is renamed to `SecretResourceDefinition`